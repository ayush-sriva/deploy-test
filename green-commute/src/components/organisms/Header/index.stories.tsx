import { ComponentStory, ComponentMeta } from '@storybook/react';
import Header from '.';

export default {
  title: 'components/organisms/Header',
  component: Header,
} as ComponentMeta<typeof Header>;

const Template: ComponentStory<typeof Header> = (args) => <Header {...args} />;

export const AppHeader = Template.bind({});

AppHeader.args = {
  userLocation: 'East Maredpally, Secunderabad',
};
