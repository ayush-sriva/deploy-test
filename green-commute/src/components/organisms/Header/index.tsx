import { AppBar, Container, Grid, styled } from '@mui/material';
import logoUrl from '../../../../public/assets/logos/logoWhite.png';
import Logo from '../../atoms/Logo';
import theme from '../../../theme';
import LocationField from '../../molecules/LocationField';
import { ChangeEvent, useState } from 'react';
import Message from '../../../../public/assets/icons/message.png';
import Avatar from '../../atoms/Avatar';
import BellUrl from '../../../../public/assets/icons/notificationIcon.png';
import avatarUrl from '../../../../public/assets/illustrations/user-img.png';

interface HeaderProps {
  userLocation: string;
}

const AppContainer = styled(Container)({
  [theme.breakpoints.up('lg')]: {
    paddingLeft: '35px !important',
    paddingRight: '35px !important',
  },
});

const MainGrid = styled(Grid)({
  paddingTop: `${theme.spacing(4)}`,
  paddingBottom: `${theme.spacing(4)}`,
});

const MainLogo = styled(Logo)({
  marginLeft: `${theme.spacing(2)}`,
  marginRight: `${theme.spacing(3)}`,
});

const Header: React.FC<HeaderProps> = ({ userLocation }) => {
  const [loc, setLocation] = useState<string>(userLocation);

  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
    setLocation(event.target.value);
    console.log('loc: ' + loc);
  };
  return (
    <AppBar elevation={0} color='secondary'>
      <AppContainer maxWidth='xl'>
        <MainGrid
          container
          rowSpacing={{ xs: 2, sm: 3, md: 4, lg: 5, xl: 6 }}
          justifyContent='space-around'
          alignItems='center'
        >
          <Grid item xs={3} lg={3} xl={3}>
            <MainLogo url={`${logoUrl}`}></MainLogo>
          </Grid>
          <Grid item xs={6} lg={6} xl={5}>
            <LocationField width='500' onChangeHandler={handleChange} location={loc}></LocationField>
          </Grid>
          <Grid item xs lg xl={4}>
            <Grid container direction='row' spacing={5} justifyContent='flex-end' alignItems='center'>
              <Grid item>
                <Logo url={`${Message}`}></Logo>
              </Grid>
              <Grid item>
                <Logo url={`${BellUrl}`}></Logo>
              </Grid>
              <Grid item>
                <Avatar alt={'user-profile'} src={`${avatarUrl}`}></Avatar>
              </Grid>
            </Grid>
          </Grid>
        </MainGrid>
      </AppContainer>
    </AppBar>
  );
};

Header.defaultProps = {
  userLocation: 'East MaredPally, Secunderabad',
};

export default Header;
