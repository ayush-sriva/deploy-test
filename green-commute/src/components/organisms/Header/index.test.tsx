import ReactTestRenderer from 'react-test-renderer';
import Header from '.';

describe('Header test', () => {
  it('renders the app header correctly', () => {
    const tree = ReactTestRenderer.create(<Header userLocation='Hyderabad' />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
