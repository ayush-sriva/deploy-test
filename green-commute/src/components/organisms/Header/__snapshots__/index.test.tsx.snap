// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test renders the app header correctly 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorSecondary MuiAppBar-positionFixed mui-fixed css-va8nif-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthXl css-9pe1zp-MuiContainer-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-1kpcvx6-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3 css-1equabv-MuiGrid-root"
      >
        <img
          alt="Logo"
          data-testid="logo"
          src="logoWhite.png"
        />
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-lg-6 MuiGrid-grid-xl-5 css-1c49gy6-MuiGrid-root"
      >
        <div
          className="MuiBox-root css-0"
        >
          <div
            className="MuiFormControl-root MuiTextField-root css-ozplz9-MuiFormControl-root-MuiTextField-root"
            data-testid="textfield-loc"
          >
            <div
              className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1ptx2yq-MuiInputBase-root-MuiInput-root"
              onClick={[Function]}
            >
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
              >
                <span
                  className="notranslate"
                >
                  â€‹
                </span>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  color="transparent"
                  data-testid="SVGIcon"
                  focusable="false"
                >
                  locationIconWhite.svg
                </svg>
              </div>
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInput-input MuiInputBase-input MuiInputBase-inputAdornedStart makeStyles-input-1 css-1x51dt5-MuiInputBase-input-MuiInput-input"
                defaultValue="Hyderabad"
                disabled={false}
                id="input-with-icon-textfield"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true MuiGrid-grid-lg-true MuiGrid-grid-xl-4 css-q0ct51-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 css-i4k12k-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="Logo"
              data-testid="logo"
              src="message.png"
            />
          </div>
          <div
            className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="Logo"
              data-testid="logo"
              src="notificationIcon.png"
            />
          </div>
          <div
            className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              className="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
            >
              <img
                alt="user-profile"
                className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="user-img.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;
